hosting/src/views/Events/Events.ts

```diff
@@ -1,40 +1,14 @@
-import dayjs, { Dayjs } from 'dayjs'
 import { setup, Vue } from 'vue-class-component'
 import { useMeta } from 'vue-meta'
 
+import { EventUi } from '@/models/event'
 import { Route } from '@/router/route-decorator'
 
-interface EventUi { // TODO: Extract to UI `models`
-  slug: string
-  name: string
-  description: string
-  dateStart: Dayjs
-  dateEnd: Dayjs
-  time: string
-  location: string
-  series?: string
-  seriesType?: string
-  badges?: string[]
-  hostId?: string
-  hostName?: string
-  registrationRequired?: boolean
-  registrationUrl?: string
-}
-
 @Route({ name: 'Events', path: '/events' })
 export default class Events extends Vue {
   meta = setup(() => useMeta({ title: 'Events' }))
 
-  get events(): EventUi[] { // TODO: Factor out as Vuex getter
-    return this.$store.state.events.map(event => {
-      const dsDayjs = dayjs.unix(event.dateStart.seconds)
-      const deDayJs = dayjs.unix(event.dateEnd.seconds)
-
-      return Object.assign(event, {
-        dateStart: dsDayjs,
-        dateEnd: deDayJs,
-        time: dsDayjs.format('ddd DD MMM, HH:mm') + '-' + deDayJs.format('HH:mm')
-      })
-    })
+  get events(): EventUi[] {
+    return this.$store.getters.events
```

Comment:
We could replace this with `mapGetters`

Reply:
I think `mapGetters` doesn't exist in Vuex 4 (for Vue 3). Seems someone has built an extension that restores them: https://github.com/greenpress/vuex-composition-helpers#typescript-mappings.

I don't know if it's worth adding unless it:
a) streamlines our code
-AND-
b) has typescript support

Reply:
CLOSED